### PR TITLE
workload: yield simulation ownership of objects outside nearby region

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

File: libraries/physics/src/EntityMotionState.h [hunk 1]
```diff
@@ -155,6 +157,7 @@ class EntityMotionState : public ObjectMotionState {
     mutable uint8_t _accelerationNearlyGravityCount;
     uint8_t _numInactiveUpdates { 1 };
     uint8_t _bidPriority { 0 };
+    uint8_t _region { workload::Region::INVALID };
```
Comment (AndrewMeadows) [inline libraries/physics/src/EntityMotionState.h]: We cache the entity's `region` inside its `EntityMotionState` whenever it changes.  This cached value is used to modify the `_bidPriority` of outgoing updates.

File: libraries/entities/src/EntityTree.cpp [hunk 2]
```diff
@@ -360,7 +360,12 @@ bool EntityTree::updateEntity(EntityItemPointer entity, const EntityItemProperti
                     // the sender is trying to take or continue ownership
                     if (entity->getSimulatorID().isNull()) {
                         // the sender is taking ownership
-                        properties.promoteSimulationPriority(RECRUIT_SIMULATION_PRIORITY);
+                        SimulationOwner owner = properties.getSimulationOwner();
+                        if (owner.getPriority() == VOLUNTEER_SIMULATION_PRIORITY) {
+                            // the entity-server always promotes VOLUNTEER to RECRUIT to avoid ownership thrash
+                            // when dynamic objects first activate and multiple participants bid simultaneously
+                            properties.promoteSimulationPriority(RECRUIT_SIMULATION_PRIORITY);
+                        }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityTree.cpp]: @samcake, this is the slight modification to the entity-server behavior: this is inside a larger `isServer()` context.  Really this is a small bug **fix**: implementing the correct intended behavior.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9324/app_13054_PR_6acd13b97fa80d7e7f51d1c7bc085686a9ecdf5b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9324/HighFidelity-Beta-PR13054-6acd13b97fa80d7e7f51d1c7bc085686a9ecdf5b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9324/HighFidelity-Beta-Interface-PR13054-6acd13b97fa80d7e7f51d1c7bc085686a9ecdf5b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9324/HighFidelity-Beta-PR13054-6acd13b97fa80d7e7f51d1c7bc085686a9ecdf5b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9324/HighFidelity-Beta-Interface-PR13054-6acd13b97fa80d7e7f51d1c7bc085686a9ecdf5b.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9337/app_13054_PR_5811123d7cacb259baa3c90f0b06ed0e84d125a6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9337/HighFidelity-Beta-PR13054-5811123d7cacb259baa3c90f0b06ed0e84d125a6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9337/HighFidelity-Beta-Interface-PR13054-5811123d7cacb259baa3c90f0b06ed0e84d125a6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9337/HighFidelity-Beta-PR13054-5811123d7cacb259baa3c90f0b06ed0e84d125a6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9337/HighFidelity-Beta-Interface-PR13054-5811123d7cacb259baa3c90f0b06ed0e84d125a6.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9368/app_13054_PR_003ea5ae3609d10b943e4afedb902c7377ac3566.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9368/HighFidelity-Beta-PR13054-003ea5ae3609d10b943e4afedb902c7377ac3566.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9368/HighFidelity-Beta-Interface-PR13054-003ea5ae3609d10b943e4afedb902c7377ac3566.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9368/HighFidelity-Beta-PR13054-003ea5ae3609d10b943e4afedb902c7377ac3566.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9368/HighFidelity-Beta-Interface-PR13054-003ea5ae3609d10b943e4afedb902c7377ac3566.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9412/app_13054_PR_af85642ad4bd1295759e400ed7cc4820fa01125f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9412/HighFidelity-Beta-PR13054-af85642ad4bd1295759e400ed7cc4820fa01125f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9412/HighFidelity-Beta-Interface-PR13054-af85642ad4bd1295759e400ed7cc4820fa01125f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9412/HighFidelity-Beta-PR13054-af85642ad4bd1295759e400ed7cc4820fa01125f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9412/HighFidelity-Beta-Interface-PR13054-af85642ad4bd1295759e400ed7cc4820fa01125f.exe

**Pull request merged**